go/arrow/array/list_test.go

```diff
@@ -211,3 +211,190 @@ func TestListArraySlice(t *testing.T) {
 		t.Fatalf("got=%q, want=%q", got, want)
 	}
 }
+
+func TestLargeListArray(t *testing.T) {
+	pool := memory.NewCheckedAllocator(memory.NewGoAllocator())
+	defer pool.AssertSize(t, 0)
+
+	var (
+		vs      = []int32{0, 1, 2, 3, 4, 5, 6}
+		lengths = []int{3, 0, 4}
+		isValid = []bool{true, false, true}
+		offsets = []int64{0, 3, 3, 7}
+	)
+
+	lb := array.NewLargeListBuilder(pool, arrow.PrimitiveTypes.Int32)
+	defer lb.Release()
+
+	for i := 0; i < 10; i++ {
+		vb := lb.ValueBuilder().(*array.Int32Builder)
+		vb.Reserve(len(vs))
+
+		pos := 0
+		for i, length := range lengths {
+			lb.Append(isValid[i])
+			for j := 0; j < length; j++ {
+				vb.Append(vs[pos])
+				pos++
+			}
+		}
+
+		arr := lb.NewArray().(*array.LargeList)
+		defer arr.Release()
+
+		arr.Retain()
+		arr.Release()
+
+		if got, want := arr.DataType().ID(), arrow.LARGE_LIST; got != want {
+			t.Fatalf("got=%v, want=%v", got, want)
```

Comment:
Just for the record, doesn't Go have a way to simplify this boilerplate? In C++ or Python, we would use custom assertions (e.g. "ASSERT_EQ") provided by the testing framework to reduce this to a simple one-liner and yet get good error diagnostics.

Reply:
It does, and you can see it used elsewhere in the tests. I just took the shortcut here to copy and paste the existing tests and shifted the type IDs to be LARGE_LIST lol. 

But you're right, I should simplify this to utilize the framework better instead. I'll do that.